src/main/java/org/broadinstitute/hellbender/utils/clipping/ReadClipper.java

```diff
@@ -488,10 +492,146 @@ protected GATKRead hardClipByReferenceCoordinates(final int refStart, final int
         if ( start > 0 && stop < read.getLength() - 1) {
             throw new GATKException(String.format("Trying to clip the middle of the read: start %d, stop %d, cigar: %s", start, stop, read.getCigar().toString()));
         }
-
         this.addOp(new ClippingOp(start, stop));
         final GATKRead clippedRead = clipRead(ClippingRepresentation.HARDCLIP_BASES);
         this.ops = null;
         return clippedRead;
     }
+
+
+
+    /**
+     * Soft clip the read to the variable region (from refStart to refStop) processing also the clipped bases
```

Comment:
The new code in this class is mostly copy pasted from the hardClip functions.  Could you add private methods that accept the clipping op to be used then refactor both hard and soft clipping operations to use that shared coded?